api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreProperties.java

```diff
@@ -0,0 +1,44 @@
+package org.openmrs.module.registrationcore.api.impl;
+
+import org.openmrs.api.APIException;
+import org.openmrs.module.registrationcore.RegistrationCoreConstants;
+import org.openmrs.module.registrationcore.api.ModuleProperties;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiProvider;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+public class RegistrationCoreProperties extends ModuleProperties implements ApplicationContextAware {
+    //TODO move getting properties related to Registration Core in this class.
+
+    private ApplicationContext applicationContext;
```

Comment:
There is one that takes a String and a Class. (And you surely know the return type.)

Anyway, no need to change this.